### DISCUSSION TITLE
Mac CGAL - Compiled code with libraries

### DISCUSSION THREAD

**Discussion by petrasvestartas** [CGAL/cgal#6524] [General]
Hi,

I would like to ask your help regarding CGAL and Mac.

## Issue Details

* I compiled code using XCode following this tutorial on Mac Big Sur 11.3:
https://stackoverflow.com/questions/27995902/linking-cgal-library-with-xcode

* Problem:
If I want to to use .dylib libraries compiled on the other Mac computer the .dylib libraries are not found.

* What files do I compile and want to use on other pc:
![image](https://user-images.githubusercontent.com/18013985/164915216-138c368b-a317-4024-9b65-82df0febacab.png)

* Why there are .dll and .dylib files together
I am doing .NET (Rhino3D C#) and C++ PInvoke (CGAL).
Therefore it is a bit hard to know where is the problem.

* What works on windows, not on MAC
On windows it is enough to compile the .dlls and copy paste the gmp and mpfr libraries.
But on mac this workflow does not work.

* Question:
Any idea how to debug / find the problem would be much appreciated.
Maybe there is a way to compile all the libraries into one .dylib file?

**Comment by lrineau** [comment]
Hi @petrasvestartas. I do not see how we can help you.

As far as I know, no CGAL developer is a macOS guru. We do test CGAL on macOS, but we have never tried to compiler on one machine, and then deploy the binaries to another machine.

As far as I understand, I think one keyword that might help you search answer on the net is `rpath`. On macOS, the handling of `rpath` is quite special. Have a look on the web... for example that page: https://blog.krzyzanowskim.com/2018/12/05/rpath-what/.
